src/test/java/itest/GraphQLIT.java

```diff
@@ -588,6 +601,426 @@ void testNodesHaveIds() throws Exception {
+                        });
+        listResp = listRespFuture4.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(
+                "list should have size 0 after deleting recordings",
+                listResp.size(),
+                Matchers.equalTo(0));
+    }
+
+    @Test
+    @Order(11)
+    public void shouldReturnArchivedRecordingsFilteredByNames() throws Exception {
+        // Check preconditions
+        CompletableFuture<JsonArray> listRespFuture1 = new CompletableFuture<>();
+        webClient
+                .get(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .send(
+                        ar -> {
+                            if (assertRequestStatus(ar, listRespFuture1)) {
+                                listRespFuture1.complete(ar.result().bodyAsJsonArray());
+                            }
+                        });
+        JsonArray listResp = listRespFuture1.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+        Assertions.assertTrue(listResp.isEmpty());
+
+        // Create a new recording
+        CompletableFuture<Void> createRecordingFuture = new CompletableFuture<>();
+        MultiMap form = MultiMap.caseInsensitiveMultiMap();
+        form.add("recordingName", TEST_RECORDING_NAME);
+        form.add("duration", "5");
+        form.add("events", "template=ALL");
+        webClient
+                .post(String.format("/api/v1/targets/%s/recordings", SELF_REFERENCE_TARGET_ID))
+                .sendForm(
+                        form,
+                        ar -> {
+                            if (assertRequestStatus(ar, createRecordingFuture)) {
+                                createRecordingFuture.complete(null);
+                            }
+                        });
+        createRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // Archive the recording
+        CompletableFuture<Void> archiveRecordingFuture = new CompletableFuture<>();
+        webClient
+                .patch(
+                        String.format(
+                                "/api/v1/targets/%s/recordings/%s",
+                                SELF_REFERENCE_TARGET_ID, TEST_RECORDING_NAME))
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpMimeType.PLAINTEXT.mime())
+                .sendBuffer(
+                        Buffer.buffer("SAVE"),
+                        ar -> {
+                            if (assertRequestStatus(ar, archiveRecordingFuture)) {
+                                archiveRecordingFuture.complete(null);
+                            } else {
+
+                                archiveRecordingFuture.completeExceptionally(
+                                        new RuntimeException("Archive request failed"));
+                            }
+                        });
+
+        archiveRecordingFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        Thread.sleep(2_000L);
```

Comment:
I think you probably want to put a sleep in between creating the active recording and archiving that recording, so that there is time for the recording to gather data before it is copied to archives. Putting the sleep here between archiving and querying for what archives exist is probably not very useful.

Reply:
I erased a big chunk of code that was retrieving to make sure the recording is archived... forgot to remove the sleep

Reply:
I think the sleep makes sense to keep, but it should be in between creating the new active recording and archiving it.

Reply:
yes